### ISSUE TITLE
[Feature request] Webseed support

### ISSUE THREAD

**Issue by EtiennePerot** [rakshasa/rtorrent#66]
Opened 6 years ago is http://libtorrent.rakshasa.no/ticket/245 - to implement web seeding in rtorrent.
Wikipedia page: http://en.wikipedia.org/wiki/BitTorrent_%28protocol%29#Web_seeding

Two conflicting specifications exist, so yes this is ambiguous. The more straightforward of the two specs is the [GetRight one](http://www.getright.com/seedtorrent.html).

This is an important part of BitTorrent's original purpose: to serve as a cheap, crowd-powered way to alleviate the load on central web servers on to everyone else in the swarm. But of course the file is initially only on a web server; the first seed has to get the file somehow. Web seeding is a proper solution to this problem, which is why I think rtorrent should support it.

**Comment by f10002483a1a2403bfcc9af1bf3efb7e** [comment]
Yes this feature will be very appreciated. Because it absence is blocking to use rtorrent everywhere.

**Comment by anoma** [comment]
Is this planned?

As far as I can ascertain rtorrent is the only remaining serious client without webseed support although to be fair some other clients have partial implementations e.g. no SSL capability

**Comment by M0ns1gn0r** [comment]
Please implement, I can't download from Internet Archive due to lack of this feature.

**Comment by Dmole** [comment]
+1

**Comment by rtasson** [comment]
+1

**Comment by manasb** [comment]
I just ran into a situation where a torrent only has web seeds and rtorrent is incapable of handling that.

**Comment by anoma** [comment]
1 year anniversary of my original request.

The world moves on and not always for the better. In an age of ever growing surveillance, webseed TLS support helps protect users.

**Comment by nickfrench4111** [comment]
I gave it a shot:
https://github.com/rakshasa/libtorrent/pull/164

**Comment by chros73** [comment]
I've also added this patch into rtorrent-ps-ch on the [feature/webseed](https://github.com/chros73/rtorrent-ps-ch/tree/feature/webseed) branch, you can try to compile it (don't forget to checkout that branch after cloning).

**Comment by spacebares** [comment]
how is this still not working???

**Comment by asclepiadae** [comment]
With development happening again thanks to some sponsorship (yay!!), is there any chance of this getting triaged in the queue as a priority new feature? @rakshasa @stickz 

As it stands now, this makes it very difficult for someone running the lovely headless rTorrent from helping preserve archive.org which is proving to be completely invaluable but under threat.

**Comment by rakshasa** [comment]
It would be a non-trivial task to implement, as we'd have to detect stalled torrents, reserve chunks to download from outside the current peer connection code, etc.

Unless the work is sponsored I don't think it will happen anytime soon.

**Comment by petesimon** [comment]
please add support for webseed, either a plugin or directly in rtorrent. thanks

**Comment by Dmole** [comment]
> sponsored

About how many $USD would it take?

**Comment by stickz** [comment]
> With development happening again thanks to some sponsorship (yay!!), is there any chance of this getting triaged in the queue as a priority new feature? [@rakshasa](https://github.com/rakshasa) [@stickz](https://github.com/stickz)
> 

If any work is required for ruTorrent, I will review any pull requests submitted. Sponsorship is active currently for ruTorrent. Contributions are being reviewed and distributed to everyone. There is no backlog right now for ruTorrent contributions. I keep my time free by spending almost all of it reviewing and testing code. This is a successful strategy, that is producing results.
